itou/asp/management/commands/gen_asp_ref_fixtures.py

```diff
@@ -0,0 +1,353 @@
+
+    OR
+
+    select one or more files to import explicitly:
+    ./manage.py gen_asp_ref_fixtures --verbosity=2 --education_levels=myfile.csv ...
+
+    Files still have to be in 'import' folder.
+
+    A 'dry-run' parameter is also available:
+    ./manage.py gen_asp_ref_fixtures --verbosity=2 --dry-run
+
+    With 'dry-run', all operations but file storage are processed.
+
+    Once generated, fixtures can be imported via:
+    ./manage.py loaddata --app asp itou/asp/fixtures/*.json
+
+    Check 'itou.asp.models' for details.
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument("--dry-run", dest="dry_run", action="store_true", help="Only print data to import")
+        parser.add_argument("--education_levels")
+        parser.add_argument("--insee_communes")
+        parser.add_argument("--insee_departments")
+        parser.add_argument("--insee_countries")
+        parser.add_argument("--measures")
+        parser.add_argument("--employer_types")
+
+    def set_logger(self, verbosity):
+        """
+        Set logger level based on the verbosity option.
+        """
+        handler = logging.StreamHandler(self.stdout)
+
+        self.logger = logging.getLogger(__name__)
+        self.logger.propagate = False
+        self.logger.addHandler(handler)
+
+        self.logger.setLevel(logging.INFO)
+        if verbosity > 1:
+            self.logger.setLevel(logging.DEBUG)
+
+    def log(self, message):
+        self.logger.debug(message)
+
+    def load_dataframe(self, path, separator=_SEP):
+        self.log(f"Creating dataframe from file: {path}")
+        df = pd.read_csv(path, sep=separator)
+        df = df.where(pd.notnull(df), None)
+        self.log(df)
+        return df
+
+    def write_fixture_file(self, path, records):
+        self.log(f"Formatted {len(records)} element(s)")
+        self.log(f"Writing JSON fixture to: {path}")
+
+        if not self.dry_run:
+            with open(path, "w") as of:
+                of.write(json.dumps(records))
+        else:
+            self.log("joking: DRY-RUN enabled")
+        self.log("Done!")
+        self.log("-" * 80)
+
+    def file_exists(self, path):
+        if os.path.isfile(path):
+            self.log(f"Found input file '{path}'")
+            return True
+        self.log("No import file found. Skipping.")
+        self.log("-" * 80)
+        return False
+
+    def gen_education_levels(self, filename="ref_niveau_formation.csv"):
+        """
+        Generate ASP education level fixture
+        """
+        path = os.path.join(_IMPORT_DIR, filename)
+
+        self.log("Importing ASP education levels:\n")
+
+        if not self.file_exists(path):
+            return
+
+        export_path = os.path.join(_FIXTURES_DIR, "asp_education_levels.json")
+        model = "asp.EducationLevel"
+        df = self.load_dataframe(path)
+        records = []
+
+        for idx, row in df.iterrows():
```

Comment:
Comme tu n'utilises pas le `idx` tu peux faire une petite simplification idiomatique :

`for _, row in df.iterrows():`

Bizarre que `black` n'ait pas grincé.

Reply:
Juste.
Résidu de copier / coller, des autres cas ou j'utilise l'index.
Et non, `black` roupille